### ISSUE TITLE
runtime error "Stateful components must return a HEEx template (~H sigil or .heex extension)"

### ISSUE THREAD

**Issue by dvic** [surface-ui/surface#494]
I'm not sure if this is already known but the latest master of LV and Surface give the following runtime error when using live components: *Stateful components must return a HEEx template (~H sigil or .heex extension)*

It looks like this was introduced in https://github.com/phoenixframework/phoenix_live_view/commit/22d4071e62abab3bf2c450b84f807c7482193418

**Comment by msaraiva** [comment]
@dvic yes, that will require a small change on our side. Thanks for reporting it!

**Comment by hubertlepicki** [comment]
@msaraiva it seems like maybe you have an idea what should be changed here to support this but didn't have time to do it yet and nobody else did either... Would it be possible to write it up here, what needs to be done, so that maybe someone else can pick it up if it's really fairly simple?

**Comment by msaraiva** [comment]
Hi @hubertlepicki!

This particular issue has already been fixed here https://github.com/surface-ui/surface/tree/ms-support-lv-0.17. However, there are still one last issue that needs to be fixed before we can release Surface `v0.7`, which should be the last version that will use our custom implementations for features that were previously not available in LV.

I'm  currently wrapping up a [PR](https://github.com/phoenixframework/phoenix_live_view/pull/1747) that introduces a unified declarative API for both Surface and LV. This is the last step required to make components from both technologies fully compatible and will allow us to remove tons of code related to those custom implementations by using the LV's new public API.

I expect to have an initial mergeable version by the end of the week so I should be back working on `v0.7` in a couple of days. In the meantime, feel free to pull the [ms-support-lv-0.17](https://github.com/surface-ui/surface/tree/ms-support-lv-0.17) branch and give it a try. There should be ~5 failing tests, all related to the use of live components. Here's an example:

```
     ** (ArgumentError) cannot convert component Surface.FunctionComponentTest.Stateful with id "stateful" to HTML.
     
     A component must always be returned directly as part of a LiveView template.
     
     For example, this is not allowed:
     
         <%= content_tag :div do %>
           <.live_component module={SomeComponent} id="myid" />
         <% end %>
     
     That's because the component is inside `content_tag`. However, this works:
     
         <div>
           <.live_component module={SomeComponent} id="myid" />
         </div>
     
     Components are also allowed inside Elixir's special forms, such as
     `if`, `for`, `case`, and friends.
     
         <%= for item <- items do %>
           <.live_component module={SomeComponent} id={item} />
         <% end %>
     
     However, using other module functions such as `Enum`, will not work:
     
         <%= Enum.map(items, fn item -> %>
           <.live_component module={SomeComponent} id={item} />
         <% end %>

```
What we need to do should be the minimum necessary to make it work with LV `v0.17` as most of that implementation will be completely removed for the next version.

In case you have any further question, just ping me here or on slack.

Cheers.

**Comment by hubertlepicki** [comment]
alright @msaraiva I will keep an eye on it and will attempt to port my app to new version whenever you have a release or pre-release ready

**Comment by msaraiva** [comment]
Fixed in #546.
